flycheck-elsa.el

```diff
@@ -46,28 +46,96 @@
+flycheck.
+
+If value is a list of strings, it will be passed unchanged as the
+`:command' argument to `flycheck-define-command-checker'."
+  :group 'flycheck-elsa
+  :type '(choice
+          (const :tag "Cask" 'cask)
+          (const :tag "Emacs" 'host)
+          (repeat 'string)))
 
-We require that the project is managed by Cask and that Elsa is
-listed as a dependency."
-  (when (and (buffer-file-name)
-             (not (seq-find (lambda (f) (string-match-p f (buffer-file-name)))
-                            flycheck-elsa-ignored-files-regexps)))
-    (when-let ((cask-file (locate-dominating-file (buffer-file-name) "Cask")))
-      (let ((bundle (cask-initialize (file-name-directory cask-file))))
-        (cask-find-dependency bundle 'elsa)))))
+(defun flycheck-elsa--filename-ignored-p (filename)
+  "Return non-nil if FILENAME is matched by one of `flycheck-elsa-ignored-files-regexps'."
+  (seq-find
+   (lambda (regexp) (string-match-p regexp filename))
+   flycheck-elsa-ignored-files-regexps))
+
+(defun flycheck-elsa--enable-p ()
+  "Return non-nil if we can enable Elsa in current buffer."
+  (if (flycheck-elsa-cask-p)
+      (when (and (buffer-file-name)
+                 (not (flycheck-elsa--filename-ignored-p (buffer-file-name))))
+        (when-let ((cask-file (locate-dominating-file (buffer-file-name) "Cask")))
+          (let ((bundle (cask-initialize (file-name-directory cask-file))))
+            (cask-find-dependency bundle 'elsa))))
+    (if (buffer-file-name)
+        (not (flycheck-elsa--filename-ignored-p (buffer-file-name)))
+      t)))
 
 (defun flycheck-elsa--working-directory (&rest _)
   "Return the working directory where the checker should run."
   (if (buffer-file-name)
-      (when-let (file (locate-dominating-file (buffer-file-name) "Cask"))
-        (file-name-directory file))
+      (if (flycheck-elsa-cask-p)
+          (when-let (file (locate-dominating-file (buffer-file-name) "Cask"))
+            (file-name-directory file))
+        (file-name-directory (cdr (project-current))))
     default-directory))
 
+(defun flycheck-elsa--cask-command ()
+  "Return a list of strings to start Elsa within a Cask project."
+  '("cask" "exec" "elsa"))
+
+(defun flycheck-elsa--elsa-dependency-directories ()
+  "Return a list of directories where Elsa and its dependencies are installed."
+  (let ((directories (mapcar
+                      (lambda (lib)
+                        (file-name-directory (find-library-name lib)))
+                      '("elsa" "dash" "trinary" "f" "s" "flycheck"))))
+    (seq-uniq directories)))
+
+(defun flycheck-elsa--host-command ()
+  "Return a list of strings to start Emacs with Elsa.
+Elsa and its dependencies are expected to be in `load-path'."
+  `("emacs"
+    ,@flycheck-emacs-args
+    ,@(mapcan
+       (lambda (directory) (list "-L" directory))
+       (flycheck-elsa--elsa-dependency-directories))
+    "--eval" "(require 'flycheck)"
+    "--load" "elsa"
```

Comment:
This logic is basically handled in the `bin/elsa` script.  Maybe we can use that instead and pass the `-L` arguments there.

It would not work on windows (which can be solved) but again gives us the control over the process from the `Elsa` project instead of having to update this one as well.

Reply:
fixed